osh/worker/csmock_runner.py

```diff
@@ -151,6 +166,13 @@ def srpm_download_analyze(self, analyzers, srpm_name, srpm_url, profile=None,
         srpm_path = self.download_file(srpm_url, srpm_name)
         return self.analyze(analyzers, srpm_path, profile, su_user, additional_arguments, **kwargs)
 
+    def dist_git_url_analyze(self, analyzers, dist_git_url, profile=None,
+                             su_user=None, additional_arguments=None, **kwargs):
+        if profile != "cspodman":
+            print("dist-git URL scan is incompatible with profile:", profile, file=sys.stderr)
+            return None, 2
+        return self.analyze(analyzers, dist_git_url, profile, su_user, additional_arguments, **kwargs)
```

Comment:
Where do we protect `dist_git_url` against shell command injection?  My quick check suggests that nowhere:
```sh
osh-cli mock-build --hub https://covscan.lab.eng.brq2.redhat.com/osh/xmlrpc --config=cspodman --git-url "https://example/ --help; ls -l ~/.config/configstore; #hash"
```

Reply:
There is a dist git URL check here: https://github.com/openscanhub/openscanhub/pull/278/files#diff-26bebd5671ea9c96d9eba3bdf8d8fcdcd3a4f8c594e359d29464c3623ff3aee3R36

But it appears the error was thrown before the validation is even triggerred. I'll see what can be done to throw this error earlier, probably to add some validation in cspodman cli.

Reply:
Validation in `osh-cli` is not a solution to the security issue because anyone can comment out the validation in their copy of `osh-cli` or just use the XML-RPC interface directly.

Reply:
... and validation in `cspodman` would not help either because the shell injection happens in the shell that runs `cspodman`, not in the implementation of `cspodman`.

Reply:
I think it's still helpful from client side because we should always raise/return earlier. I had this commit addressed: https://github.com/openscanhub/openscanhub/pull/278/commits/69a902a156b4ab0b816a6bb15b81c62423094848

I'll see how similar sanitation/validation can be done from hub too. working in progress.